qiskit/circuit/library/standard_gates/equivalence_library.py

```diff
@@ -160,6 +161,27 @@
 def_r.append(U3Gate(theta, phi - pi / 2, -phi + pi / 2), [q[0]])
 _sel.add_equivalence(RGate(theta, phi), def_r)
 
+# IGate
+q = QuantumRegister(1, "q")
+def_id = QuantumCircuit(q)
+def_id.append(UGate(0, 0, 0), [q[0]])
+_sel.add_equivalence(IGate(), def_id)
+
+q = QuantumRegister(1, "q")
+def_id_rx = QuantumCircuit(q)
+def_id_rx.append(RXGate(0), [q[0]])
+_sel.add_equivalence(IGate(), def_id_rx)
+
+q = QuantumRegister(1, "q")
+def_id_ry = QuantumCircuit(q)
+def_id_ry.append(RYGate(0), [q[0]])
+_sel.add_equivalence(IGate(), def_id_ry)
+
+q = QuantumRegister(1, "q")
+def_id_rz = QuantumCircuit(q)
+def_id_rz.append(RZGate(0), [q[0]])
+_sel.add_equivalence(IGate(), def_id_rz)
```

Comment:
```suggestion
q = QuantumRegister(1, "q")
def_id = QuantumCircuit(q)
_sel.add_equivalence(IGate(), def_id)
```

I think this is the same as @jakelishman 's question, but is there a reason to replace `IGate` with other effective nop gates as opposed to replacing with an empty circuit? If we had a subsequent pass in optimization to remove nops, then I think it would be easier to leave `IGate`'s in place here (and add `'id'` to the `BasisTranslator`'s `basic_instrs`) and find them later, as opposed to having to inspect gates in optimization to see which are effective no-ops (which we do anyway in synthesis, but is computationally harder).

I suppose one argument against this is that removing `IGate`s here might be considered an optimization, which we wouldn't want in the `BasisTranslator` or in e.g. `optimization_level=0`, but that `IGate`s also aren't supported natively on all devices. However, I think this requires some thought as not all of these nop implementations will have the same cost/behavior on devices. e.g. `UGate(0,0,0)` will on IBM devices be implemented with two single qubit `sx` gates, while `RZGate(0)` will require none.

Reply:
I'm still in favour of this definition that Kevin's suggesting above, but I also in principle don't hugely mind leaving the definitions that are already in the PR.

Fwiw, the level of basis translation doesn't / shouldn't resolve or know the number of pulses used to implement a device, so I wouldn't call a translation of `IGate` to the empty body an "optimisation", I'd just call it a valid translation to something a backend can run.  If the backend natively supports `IGate`, no translation occurs.  It's up to pulse-aware passes to resynthesise 1q / 2q blocks into pulse-optimal sequences, if the user chooses that.

Reply:
Following the discussion in https://github.com/Qiskit/qiskit/pull/7403#pullrequestreview-1537269508 , the code as written looks to be the right approach.